core/src/main/java/com/linecorp/armeria/server/ServiceWithPathMapping.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server;
+
+import java.util.List;
+
+import com.linecorp.armeria.common.Request;
+import com.linecorp.armeria.common.Response;
+
+/**
+ * An interface that enables getting {@link PathMapping} of a {@link Service}.
+ *
+ * @param <I> the {@link Request} type
+ * @param <O> the {@link Response} type
+ */
+public interface ServiceWithPathMapping<I extends Request, O extends Response> extends Service<I, O> {
+    /**
+     * Returns the list of {@link PathMapping} to which this {@link Service} is bound.
+     */
+    List<PathMapping> pathMappings();
```

Comment:
How about returning `Set<PathMapping>`? They are not expected to have any duplicates.